nameserver/names.go

```diff
@@ -0,0 +1,28 @@
+package nameserver
+
+import (
+	"strings"
+
+	"github.com/miekg/dns"
+)
+
+// Get the number of components of a name
+// For example, "redis" is 1, "redis.weave.local"  is 3...
+func nameNumComponents(name string) int {
+	if len(name) == 0 || name == "." {
+		return 0
+	}
+	return strings.Count(dns.Fqdn(name), ".")
+}
+
+// normalize a name by
+// a) adding the local domain when it is a single component name
+// b) converting the result to FQDN
+func nameNormalize(n string, domain string) string {
```

Comment:
`normalizeName` would break the scheme where all functions in `names.go` have the form `name*`...